src/Discord.Net/Entities/Roles/IRole.cs

```diff
@@ -0,0 +1,22 @@
+using System.Threading.Tasks;
+using Wumpus.Entities;
+
+namespace Discord
+{
+    public interface IRole : ISnowflakeEntity
+    {
+        ValueTask<IGuild> GetGuildAsync(StateBehavior stateBehavior = default, RequestOptions? options = default);
+
+        string Name { get; }
+        Color Color { get; }
+        int Position { get; }
+
+        bool IsHoisted { get; }
+        bool IsManaged { get; }
+        bool IsMentionable { get; }
+
+        GuildPermissions Permissions { get; } // review: use wump permissions directly or wrap?
```

Comment:
The prior permissions wrapper didn't bug me all that much, but it didn't strike me as that great either. Using the raw `enum` seems much simpler and much more approachable for consumers (as it's a core language feature). Discord.NET can always add some extension methods for reading/manipulating the values if desired.